### PR TITLE
Add RSA 2048/4096 "raw" signing support

### PR DESCRIPTION
**Pull request by mtvaught-ibm**
Some existing hardware requires a custom DigestInfo/padding scheme.
Instead of supporting all of these, the signer can now pre-format
the padding on their hash and pass it as a raw payload to the
framework.

By doing a raw signing, it is up to the signer to verify that their
raw payload is correct before sending it to the server and verifying
that the signature can be successfully validated. The framework can
only decrypt the signature with the public key and compare it to the
input raw payload.

### CODE REVIEW

**Comment by mtvaught-ibm** [comment]
Please review carefully.

File: src/signframework/.gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+*.log
```
Comment (cjengel) [inline src/signframework/.gitignore]: Is this a duplicate of what is in the base .gitignore?

File: src/signframework/ossl_functions.c [hunk 2]
```diff
@@ -1060,9 +1060,143 @@ long osslVerifyRSA512(int *valid,			/* output boolean */
...
+
+/* osslVerifyRaw() verifies the digital 'signature' over 'digest' using the public key modulus
+   'nArray' and exponent 'eArray'.  'digest' the SHA-384 digest of the data.
+
+   The modulus and exponent are pure binary streams, with no formatting envelope.
+*/
+
+long osslVerifyRaw(int *valid,			/* output boolean */
+		   unsigned char *rawPayload,
+           unsigned long rawPayloadSize,
```
Comment (cjengel) [inline src/signframework/ossl_functions.c]: there appear to be a bunch of alignment issues throughout

File: src/signframework/ossl_functions.c [hunk 3]
```diff
@@ -1060,9 +1060,143 @@ long osslVerifyRSA512(int *valid,			/* output boolean */
...
+
+long osslVerifyRSARaw(int *valid,			/* output boolean */
+		      unsigned char *rawPayload,
+              unsigned long rawPayloadSize,
+		      RSA *rsaPubKey, 		/* OpenSSL RSA key token */
+		      unsigned char *signature,
+		      unsigned long signature_size)
+{
+    long 		rc = 0;			/* function return code */
+    int			irc;			/* OpenSSL return code */
```
Comment (cjengel) [inline src/signframework/ossl_functions.c]: init stack

File: src/signframework/ossl_functions.c [hunk 4]
```diff
@@ -1060,9 +1060,143 @@ long osslVerifyRSA512(int *valid,			/* output boolean */
...
+           unsigned long rawPayloadSize,
+		   unsigned char *eArray,
+		   unsigned long eLength,
+		   unsigned char *nArray,
+		   unsigned long nLength,
+		   unsigned char *signature,
+		   unsigned long signature_size)
+{
+    long 		rc = 0;			/* function return code */
+    RSA *		rsaPubKey;		/* public key in OpenSSL structure format */
```
Comment (cjengel) [inline src/signframework/ossl_functions.c]: init stack

**Pull request merged**